server/src/internalClusterTest/java/org/opensearch/remotestore/CompositeDirectoryIT.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.remotestore;
+
+import org.apache.lucene.store.Directory;
+import org.apache.lucene.store.FilterDirectory;
+import org.opensearch.action.admin.indices.get.GetIndexRequest;
+import org.opensearch.action.admin.indices.get.GetIndexResponse;
+import org.opensearch.cluster.metadata.IndexMetadata;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.common.util.FeatureFlags;
+import org.opensearch.index.IndexModule;
+import org.opensearch.index.IndexService;
+import org.opensearch.index.shard.IndexShard;
+import org.opensearch.index.store.CompositeDirectory;
+import org.opensearch.indices.IndicesService;
+import org.opensearch.indices.replication.common.ReplicationType;
+
+import static org.opensearch.cluster.metadata.IndexMetadata.SETTING_REMOTE_SEGMENT_STORE_REPOSITORY;
+import static org.opensearch.cluster.metadata.IndexMetadata.SETTING_REMOTE_STORE_ENABLED;
+import static org.opensearch.cluster.metadata.IndexMetadata.SETTING_REMOTE_TRANSLOG_STORE_REPOSITORY;
+import static org.opensearch.cluster.metadata.IndexMetadata.SETTING_REPLICATION_TYPE;
+import static org.opensearch.test.hamcrest.OpenSearchAssertions.assertAcked;
+
+public class CompositeDirectoryIT extends RemoteStoreBaseIntegTestCase {
+    public void testCompositeDirectory() throws Exception {
```

Comment:
Is this test case sufficient ? 
Do we need to add tests for lets say listAll (when some files are completely on remote, some blocks are local and some files are completely newly created)?
Can we explore [BaseDirectoryTestCase](https://github.com/apache/lucene/blob/main/lucene/test-framework/src/java/org/apache/lucene/tests/store/BaseDirectoryTestCase.java) in Lucene as well

Reply:
This is a IT test which was for basic sanity testing, will add unit tests and strengthen this IT test as well in upcoming revisions